src/interceptors/WebSocket/WebSocketServerConnection.ts

```diff
@@ -28,56 +28,32 @@ export class WebSocketServerConnection {
     this[kEmitter] = new EventTarget()
     this.mockCloseController = new AbortController()
 
-    // Handle incoming events from the actual server.
-    // The (mock) WebSocket instance will call this
-    // whenever a "message" event from the actual server
-    // is dispatched on it (the dispatch will be skipped).
-    this.transport.onIncoming = (event) => {
-      // Clone the event to dispatch it on this class
-      // once again and prevent the "already being dispatched"
-      // exception. Clone it here so we can observe this event
-      // being prevented in the "server.on()" listeners.
-      const messageEvent = bindEvent(
-        this.realWebSocket!,
-        new CancelableMessageEvent('message', {
-          data: event.data,
-          origin: event.origin,
-          cancelable: true,
-        })
-      )
-
-      /**
-       * @note Emit "message" event on the WebSocketClassServer
-       * instance to let the interceptor know about these
-       * incoming events from the original server. In that listener,
-       * the interceptor can modify or skip the event forwarding
-       * to the mock WebSocket instance.
-       */
-      this[kEmitter].dispatchEvent(messageEvent)
-
-      /**
-       * @note Forward the incoming server events to the client.
-       * Preventing the default on the message event stops this.
-       */
-      if (!messageEvent.defaultPrevented) {
-        this.socket.dispatchEvent(
-          bindEvent(
-            /**
-             * @note Bind the forwarded original server events
-             * to the mock WebSocket instance so it would
-             * dispatch them straight away.
-             */
-            this.socket,
-            // Clone the message event again to prevent
-            // the "already being dispatched" exception.
-            new MessageEvent('message', {
-              data: event.data,
-              origin: event.origin,
-            })
-          )
-        )
+    // Automatically forward outgoing client events
+    // to the actual server unless the outgoing message event
+    // has been prevented. The "outgoing" transport event it
+    // dispatched by the "client" connection.
+    this.transport.addEventListener('outgoing', (event) => {
```

Comment:
## Feature

The actual implementation for the client-to-server automatic forwarding. The `client` connection results in the "outgoing" transport event. We listen to it in the `server` connection, and decide when to forward it to the server.